### PR TITLE
feat: added support for ARRAY_MATCH filter grouping

### PR DESCRIPTION
**Pull request by sampotts**

### CODE REVIEW

**Review by tuanddd — APPROVED** [review]
LGTM, just push a new doc section for `ARRAY_MATCH` and you're good to go.

File: packages/hooks/src/ContextProvider/controllers/filters/utils.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    ) as Record<string, Array<string>>;
+
+  // Build filter expression
+  if (type === 'filter') {
+    // Build the list of non grouped filters
+    const basicFilters = list.filter(({ group, count }) => !group && !count).map(({ expression }) => `(${expression})`);
+
+    return Object.entries(groups)
+      .reduce((out, [group, expressions]) => {
+        if (group !== 'undefined' && !isEmpty(expressions)) {
```
Comment (zlatanpham) [inline packages/hooks/src/ContextProvider/controllers/filters/utils.ts:47]: Should it be `typeof group !== 'undefined'`?

**Review by zlatanpham — APPROVED** [review]
Aside from one minor comment, it looks good 👍

File: packages/hooks/src/ContextProvider/controllers/filters/utils.ts (hunk 1 above)
Comment (sampotts) [reply packages/hooks/src/ContextProvider/controllers/filters/utils.ts:47]: Good catch but due to the way the groupBy function works, if `group` is `undefined` in the `FilterBuilder` then it gets turned into a string for the key of the object so it ends up as `'undefined'` instead. Hopefully, that makes sense 👍

**Pull request merged**